File: app/src/main/java/com/example/news/util/EntityMapper.kt
package com.example.news.util

/**
 * Interface for mapping between entity model and domain model
 * @param EntityModel the entity model type
 * @param DomainModel the domain model type
 */
interface EntityMapper<EntityModel, DomainModel> {

    /**
     * Maps from entity model to domain model
     * @param entity the entity model
     * @return the domain model
     */
    fun toDomain(entityModel: EntityModel): DomainModel

    /**
     * Maps from domain model to entity model
     * @param domainModel the domain model
     * @return the entity model
     */
    fun toEntity(domainModel: DomainModel): EntityModel
}

File: news/data/cache/EntityMapper.swift
//
//  EntityMapper.swift
//  news
//
//  Created by Mendez, Juan on 9/16/25.
//

import Foundation

/**
 * Interface for mapping between entity model and domain model
 * @param EntityModel the entity model type
 * @param DomainModel the domain model type
 */
protocol EntityMapper where EntityModel: Sendable, DomainModel: Sendable {
    associatedtype EntityModel
    associatedtype DomainModel

    /// Maps from entity model to domain model
    func toDomain(_ entityModel: EntityModel) -> DomainModel

    /// Maps from domain model to entity model
    func toEntity(_ domainModel: DomainModel) -> EntityModel
}
